packages/principles-core/src/runtime-v2/feedback/__tests__/safe-stringify.test.ts

```
@@ -0,0 +1,200 @@
+    const result = safeStringifyPreview(obj);
+    expect(result).toBe('{"name":"test","count":10}');
+  });
+
+  it('stringifies arrays', () => {
+    const arr = [1, 2, 3];
+    const result = safeStringifyPreview(arr);
+    expect(result).toBe('[1,2,3]');
+  });
+
+  it('handles BigInt values', () => {
+    const obj = { big: BigInt(12345678901234567890) };
```

ESLint

[error] This number literal will lose precision at runtime.

(no-loss-of-precision)